src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -204,7 +208,7 @@ public String getResourceURI(IResource resource) {
 		if(ResourceKind.LIST.equals(resource.getKind())) throw new UnsupportedOperationException("Generic create operation not supported for resource type 'List'");
 		try {
 			namespace = ResourceKind.PROJECT.equals(resource.getKind()) ? "" : namespace;
```

Comment:
Shouldnt we use "null" instead of "" as a namespace if we dont use it? I see that URLBuilder will accept both (empty or null - it's using StringUtils#isBlank). But in my understanding null is more obvious than an empty string for stating that the namespace shall not be used.